### ISSUE TITLE
Question: Is there support for debug logging ? (log messages only when compile in debug mode)

### ISSUE THREAD

**Issue by lock9** [neo-project/neo-compiler#146]
Hello,
I was wondering if it is too difficult to have some sort of log instruction that is only added to the code if we compile it using a debug flag? Does it exist already?

The problem is:
I use logs to do some tests, but I don't need those logs when my SC is compiled in production mode. 
Is there a way to do some kind of "ifdef" to avoid these  instructions when I compile it?

Thanks

**Comment by igormcoelho** [comment]
You mean in C#? Can you detect this "Debug" situation in any C# flag? If you manage that, I think it's just a matter of creating a function that only executes in Debug situation, and we don't even need to change compiler for that.

**Comment by lock9** [comment]
It's because I debug  / test the .AVM, not the C# code.
Do you think I should use #ifdef with Runtime.Log?

**Comment by igormcoelho** [comment]
Yeah, I think it's a good idea. I tried to replace `DEBUG` with `true` and code appeared. The point is that neocompiler.io uses Release.. I may need to test locally to see if DEBUG is automatically invoked by Debug configuration, so we could add that to our interface too. But perhaps you can try easily using Visual Studio or any other IDE.
```cs
        #if DEBUG
        static void DebugLog(string log) { 
            Runtime.Log(log);
        }
        #endif         
```
